src/Spanner/Session/CacheSessionPool.php

```diff
@@ -777,4 +764,63 @@ private function validateConfig()
             );
         }
     }
+
+    /**
+     * Delete the provided sessions.
+     *
+     * @param array $sessions
+     */
+    private function deleteSessions(array $sessions)
+    {
+        $calls = [];
+
+        foreach ($sessions as $session) {
+            $calls[] = $this->database->connection()
+                ->deleteSessionAsync([
+                    'name' => $session['name'],
+                    'database' => $this->database->name()
+                ]);
+        }
+
+        foreach ($calls as $call) {
+            $call->wait();
```

Comment:
OK, no worries. If in fact you find that it is necessary (or decide that even though not necessary, it is desirable) what do you think about the following: have the `deleteSessions` method return the list of UnaryCalls, or an object (like a promise) that wraps them, and then after calling `deleteSessions`, register the need to call `wait` before completing the request (this could be with `register_shutdown_function`, or it could be completely contained within the client, so that a method such as [`Database::snapshot`](https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/src/Spanner/Database.php#L545) would check and call `wait` on the `deleteSession` calls after the call to `$this->operation->snapshot` has completed.